packages/wrangler/src/deployment-bundle/create-worker-upload-form.ts

```diff
@@ -125,6 +125,12 @@ export type WorkerMetadataBinding =
 	  }
 	| { type: "mtls_certificate"; name: string; certificate_id: string }
 	| { type: "pipelines"; name: string; pipeline: string }
+	| {
+			type: "secret_store";
+			name: string;
+			store_id: string;
+			secret_name: string;
+	  }
```

Comment:
Please make sure this matches the API and adjust it as you need. I made this up simply based on the shape of the secret store config.